src/main/kotlin/ua/anastasiia/finesapp/controller/FineController.kt

```diff
@@ -1,34 +1,96 @@
 import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
+import ua.anastasiia.finesapp.dto.CarRequest
 import ua.anastasiia.finesapp.dto.FineRequest
-import ua.anastasiia.finesapp.dto.FineResponse
-import ua.anastasiia.finesapp.entity.Fine
-import ua.anastasiia.finesapp.service.FineService
+import ua.anastasiia.finesapp.dto.TotalFineSumResponse
+import ua.anastasiia.finesapp.dto.TrafficTicketRequest
+import ua.anastasiia.finesapp.entity.MongoFine
+import ua.anastasiia.finesapp.service.MongoFineService
+import java.time.LocalDate
 
 @RestController
 @RequestMapping(value = ["/fines"])
-class FineController(val fineService: FineService) {
+@Suppress("TooManyFunctions")
+class FineController(val mongoFineService: MongoFineService) {
 
     @GetMapping
-    fun getAllFines(): List<FineResponse> = fineService.getAllFines()
+    fun getAllFines(): List<MongoFine> = mongoFineService.getAllFines()
+
+    @GetMapping("location")
+    fun getAllFinesInLocation(@RequestParam longitude: Double, @RequestParam latitude: Double): List<MongoFine> =
+        mongoFineService.getAllFinesInLocation(longitude, latitude)
+
+    @GetMapping("date/{date}")
+    fun getAllFinesInLocation(@PathVariable date: LocalDate): List<MongoFine> =
+        mongoFineService.getAllFinesByDate(date)
+
+    @GetMapping("id/{id}")
+    fun getFineById(@PathVariable id: ObjectId): MongoFine =
+        mongoFineService.getFineById(id)
+
+    @GetMapping("plate/{plate}")
+    fun getFineByCarPlate(@PathVariable plate: String): MongoFine =
+        mongoFineService.getFineByCarPlate(plate)
 
     @PostMapping
-    fun newFine(@RequestBody fineRequest: FineRequest): Fine = fineService.createFine(fineRequest)
+    fun saveFine(@Valid @RequestBody fineRequest: FineRequest): MongoFine = mongoFineService.saveFine(fineRequest)
+
+    @PostMapping("all")
+    fun saveFines(@Valid @RequestBody mongoFines: List<FineRequest>): List<MongoFine> =
+        mongoFineService.saveFines(mongoFines)
+
+    @DeleteMapping("{id}")
+    fun deleteFineById(@PathVariable id: ObjectId): MongoFine = mongoFineService.deleteFineById(id)
+
+    @PatchMapping("add_ticket/{carPlate}")
+    fun addTrafficTicketByCarPlate(
+        @PathVariable carPlate: String,
+        @Valid @RequestBody ticketRequest: TrafficTicketRequest
+    ): MongoFine = mongoFineService.addTrafficTicketByCarPlate(carPlate, ticketRequest)
+
+    @PatchMapping("update_ticket/{carPlate}/{ticketId}")
+    fun updateTrafficTicketByCarPlateAndId(
+        @PathVariable carPlate: String,
+        @PathVariable ticketId: ObjectId,
+        @Valid @RequestBody ticketRequest: TrafficTicketRequest
+    ): MongoFine = mongoFineService.updateTrafficTicketByCarPlateAndId(carPlate, ticketId, ticketRequest)
+
+    @PatchMapping("add_violation/{carPlate}/{ticketId}/{violationIds}")
```

Comment:
```suggestion
    @PatchMapping("add_violation/car/{carPlate}/ticket/{ticketId}/violations/{violationIds}")
```
Having 3 ids in a row makes your url messy. use resource identifiers to make it more readable. applicable to all similar places